pkg/config/structure/unmarshal_test.go

```diff
@@ -814,6 +784,443 @@ feature:
+
+		assert.Equal(t, feature.Enabled, want, "unexpected marshal value, want: %s got: %s", want, feature.Enabled)
+	})
+
+	t.Run("yaml flow", func(t *testing.T) {
+		feature := struct {
+			Enabled string `yaml:"enabled,flow"`
+		}{}
+
+		err := UnmarshalKey(mockConfig, "feature", &feature)
+		assert.NoError(t, err, "%s failed to marshal: %s", "yaml flow flag", err)
+
+		assert.Equal(t, feature.Enabled, want, "unexpected marshal value, want: %s got: %s", want, feature.Enabled)
+	})
+
+	t.Run("yaml inline", func(t *testing.T) {
+		feature := struct {
+			Enabled string `yaml:"enabled,inline"`
+		}{}
+
+		err := UnmarshalKey(mockConfig, "feature", &feature)
+		assert.NoError(t, err, "%s failed to marshal: %s", "yaml inline flag", err)
+
+		assert.Equal(t, feature.Enabled, want, "unexpected marshal value, want: %s got: %s", want, feature.Enabled)
+	})
+
+	t.Run("yaml multiple flags", func(t *testing.T) {
+		feature := struct {
+			Enabled string `yaml:"enabled,inline,flow"`
+		}{}
+
+		err := UnmarshalKey(mockConfig, "feature", &feature)
+		assert.NoError(t, err, "%s failed to marshal: %s", "yaml multiple flags", err)
+
+		assert.Equal(t, feature.Enabled, want, "unexpected marshal value, want: %s got: %s", want, feature.Enabled)
+	})
+
+	t.Run("mapstructure remain", func(t *testing.T) {
+		feature := struct {
+			Enabled string `mapstructure:"enabled,remain"`
+		}{}
+
+		err := UnmarshalKey(mockConfig, "feature", &feature)
+		assert.NoError(t, err, "%s failed to marshal: %s", "mapstructure omitempty flag", err)
+
+		assert.Equal(t, feature.Enabled, want, "unexpected marshal value, want: %s got: %s", want, feature.Enabled)
+	})
+
+	t.Run("mapstructure squash errors without option", func(t *testing.T) {
+		feature := struct {
+			Enabled string `mapstructure:"enabled,squash"`
+		}{}
+
+		err := UnmarshalKey(mockConfig, "feature", &feature)
+		assert.Error(t, err)
+		assert.Contains(t, err.Error(), "EnableSquash")
+	})
+
+	t.Run("mapstructure omitempty", func(t *testing.T) {
+		feature := struct {
+			Enabled string `mapstructure:"enabled,omitempty"`
+		}{}
+
+		err := UnmarshalKey(mockConfig, "feature", &feature)
+		assert.NoError(t, err, "%s failed to marshal: %s", "mapstructure omitempty flag", err)
+
+		assert.Equal(t, feature.Enabled, want, "unexpected marshal value, want: %s got: %s", want, feature.Enabled)
+	})
+
+	t.Run("mapstructure multiple flags", func(t *testing.T) {
+		feature := struct {
+			Enabled string `mapstructure:"enabled,remain,omitempty"`
+		}{}
+
+		err := UnmarshalKey(mockConfig, "feature", &feature)
+		assert.NoError(t, err, "%s failed to marshal: %s", "mapstructure multiple flags", err)
+
+		assert.Equal(t, feature.Enabled, want, "unexpected marshal value, want: %s got: %s", want, feature.Enabled)
+	})
+
+	t.Run("mapstructure squash multiple flags errors without option", func(t *testing.T) {
+		t.Skip("FIXME")
+
+		feature := struct {
+			Enabled string `mapstructure:"enabled,remain,squash"`
```

Comment:
we don't currently split on multiple tags but maybe we should?

Reply:
Yeah, that should get fixed in a future PR. It's not urgent because I don't believe anything uses multiple tags, but it would be good to future proof it.

There's a TODO in the current implementation: https://github.com/DataDog/datadog-agent/blob/8d63f098a38301dca5db05bb6ef3845b83ce7f4b/pkg/config/structure/unmarshal.go#L337